### PR TITLE
Address iOS compiler and linter warnings

### PR DESCRIPTION
**Pull request by sixten**
This branch attempts to clear out unnecessary warnings from the iOS build.

- Merge changes from `main`.
- Update project settings and remove some unnecessary build phases.
- Address a few compiler warnings.
- Address quite a few linter warnings. Some are actively mitigated, and some silenced.

I won't be hurt if some of these changes need further work to meet project style or anything.

There's some testing I still want to do on a couple of these, but I wanted you to be able to review things sooner rather than later.

### CODE REVIEW

File: apple/Omnivore.xcodeproj/project.pbxproj
Comment (satindar) [inline apple/Omnivore.xcodeproj/project.pbxproj:964]: Neat. Didn't know this was an option here.

Comment (sixten) [reply apple/Omnivore.xcodeproj/project.pbxproj:964]: It was added in Xcode 14.2 (IIRC), along with a related warning for any build phases that didn't have input files that could be used for dependency analysis.

**Review by satindar — APPROVED** [review]
Thanks so much for this PR @sixten 🙌🏽 

I did have to disable the thread sanitizer to get it to run. Couldn't understand the error:

`Omnivore(64232,0x1058a0580) malloc: nano zone abandoned due to inability to preallocate reserved vm space.`

Possibly related to Intercom since some logging for that was also there. 

@jacksonh I'll leave it to you to merge to your branch since you'll have to update to Xcode 14.2 once this is merged into it. You can disable the thread sanitizer for now (by selecting Edit Scheme):

Screen Shot 2023-02-20 at 8 48 42 PM

**Comment by sixten** [comment]
> I did have to disable the thread sanitizer to get it to run.

Does that prevent the app from running for you? I'd noticed the message in the console output, but hadn't realized it wasn't happening before. The app still ran for me, though. (In the simulator, anyway.)

It looks like that error is [specifically related to enabling the Thread Sanitizer](https://stackoverflow.com/q/64126942/16892), but shouldn't be fatal.

> you'll have to update to Xcode 14.2 once this is merged into it.

Did the updates I made to the project prevent it from opening in previous Xcode? If so, that was definitely _not_ deliberate, and I can revert them. Were you getting an error message?

**Comment by satindar** [comment]
> > I did have to disable the thread sanitizer to get it to run.
> 
> Does that prevent the app from running for you? I'd noticed the message in the console output, but hadn't realized it wasn't happening before. The app still ran for me, though. (In the simulator, anyway.)
> 
> It looks like that error is [specifically related to enabling the Thread Sanitizer](https://stackoverflow.com/q/64126942/16892), but shouldn't be fatal.
> 
> > you'll have to update to Xcode 14.2 once this is merged into it.
> 
> Did the updates I made to the project prevent it from opening in previous Xcode? If so, that was definitely _not_ deliberate, and I can revert them. Were you getting an error message?

I saw the change in the prod file that referenced 14.2 so was assuming that it required an update. So didn't actually test it with the previous version I had but yeah you're right it should be backwards compatible for a point release. 

I'm also not running into the same error with teh thread sanitizer enabled now. It wasn't crashing before. Just logging the messages and then the simulator was stuck on a black screen 🤷🏽‍♂️

**Comment by jacksonh** [comment]
NIce, I'll merge this one and then deal with any conflicts on my branch. Awesome work.

**Pull request merged**
